examples/server/server.cpp

```
@@ -1763,12 +1763,16 @@ struct server_response {
             std::unique_lock<std::mutex> lock(mutex_results);
             condition_results.wait(lock, [&]{
                 if (!running) {
-                    SRV_DBG("%s : queue result stop\n", __func__);
-                    std::terminate(); // we cannot return here since the caller is HTTP code
+                    return true;
                 }
                 return !queue_results.empty();
             });
-
+            if (!running) {
+                SRV_DBG("%s : queue result stop\n", __func__);
+                auto res = std::make_unique<server_task_result_error>();
```

I don't think it's a good idea to return `server_task_result_error`

Some handlers create task with type cancel upon receiving `server_task_result_error`, so the server will probably hangs. You should check all code paths, not just one single use case.

It's better to throw an exception here

---

> It's not a good idea to return `server_task_result_error`
> 
> Some handlers create task with type cancel upon receiving `server_task_result_error`, so the server will probably hangs. You should check all code paths, not just one single use case.
> 
> It's better to throw an exception here


i read the code again, and i really wonder 
at this situation, the server is shutting down,   the `running` flag is set to false 
why and where the server will hang ?
If you have time, could you please point out what will cause the problem you are talking about?

@ngxson  thanks